### ISSUE TITLE
ICE: "scatterFunc != NULL".

### ISSUE THREAD

**Issue by Vsevolod-Livinskij** [ispc/ispc#1762]
ISPC fails with 
>$ ispc func.ispc
func.ispc:6:9: Error: Assertion failed (/ispc/src/ctx.cpp:2659): "scatterFunc != NULL".
        e[b][c][d] = 0;

```c++
bool e[4][20][7];
void a() {
  for (int16 b;;)
    for (int8 c;;)
      foreach (d = 0...16)
        e[b][c][d] = 0;
}
```

ISPC version:
1.14.0dev (build commit 1da899442b7af430 @ 20200520, LLVM 10.0.0)

**Comment by DeepakRajendrakumaran** [comment]
Can you please let me know what compiler switches you are using?

https://godbolt.org/z/pRtVtC - doesn't seem to be reproducible.

**Comment by dbabokin** [comment]
I can reproduce it with ``--target=avx512skx-i32x16``

**Comment by DeepakRajendrakumaran** [comment]
Thanks. Got it.

**Comment by DeepakRajendrakumaran** [comment]
Mask change for avx512 resulted in us trying find scatter for i1, Reduced case -https://godbolt.org/z/qZWLuh

**Comment by DeepakRajendrakumaran** [comment]
In theory theory, this should have been fixed with bool i8 storage change. It didn't due to a bug in the original commit.

Tentative fix - https://github.com/DeepakRajendrakumaran/ispc/commits/1762_1763

**Comment by dbabokin** [comment]
Fixed by #1766
